tests/unit/test_attach.py

```diff
@@ -18,4 +18,55 @@ def test_memray_attach_aggregated_without_output_file(
             main(["attach", "--aggregate", "1234"])
 
         captured = capsys.readouterr()
+        print("Error", captured.err)
         assert "Can't use aggregated mode without an output file." in captured.err
+
```

Comment:
@godlygeek I assume testing these options may be too cumbersome but maybe we should do some smoke testing at least....

Reply:
In the first version of this we weren't logging anything when detaching, and so it was essentially impossible to prove that the new code was reached - but now that we're logging, I suppose we could do some simple smoke tests that just check the stdio of the tracked process looking for "heap size has reached" and "seconds have elapsed". Think that's good enough?

Reply:
Yeah, that will certainly work

Reply:
Check [de94a39](https://github.com/bloomberg/memray/pull/458/commits/de94a39591fd4f945f43dd98a3ed3256ef37fd0e)

Reply:
Well, that doesn't pass CI 😅 

The more I think about this, the more I become convinced that this should be a separate subcommand instead. Rather than having a `memray attach --stop-tracking` that's mutually exclusive with almost every other `memray attach` argument, we should just have a separate `memray detach` command. At first I thought that wouldn't make sense because `memray detach` would still need to _attach_ to the process in order to uninstall the old tracker, but I think that's just me letting my knowledge of the underlying architecture and communication channels cloud my judgment. I think that a typical user would look at `memray attach` as attaching _Memray_ to the process, as opposed to gdb or lldb, and so users would find it perfectly intuitive for there to be a `memray detach` command that removes Memray from the process.

Let me know if you agree - if you do, I'll do the legwork of splitting this into a separate `memray detach` subcommand.